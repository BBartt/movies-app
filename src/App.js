import React from "react";
import { routes } from "routes/routes";
import { HashRouter, Route, Switch, Redirect } from "react-router-dom";
import GlobalStyles from "theme/GlobalStyle";
import StyledAppName from "components/atoms/StyledAppName/StyledAppName";
import NavBar from "components/organisms/NavBar/NavBar";
import Container from "components/organisms/Container/Container";
import Main from "components/templates/Main/Main";
import MoviesCards from "components/templates/MoviesCards/MoviesCards";
import MovieDetails from "components/organisms/MovieDetails/MovieDetails";
import Contest from "components/organisms/Contest/Contest";
import About from "components/organisms/About/About";
import Contact from "components/organisms/Contact/Contact";
import Footer from "components/organisms/Footer/Footer";
import {VerticleButton as ScrollUpButton} from "react-scroll-up-button";

const App = () => (
  <HashRouter>
    <>
      <GlobalStyles />
      <StyledAppName />
      <NavBar />
      <Route path={routes.movies_app} component={Container} />
      <Route path={routes.movies_app} component={Main} />

      <Switch>
        <Route
          exact
          path={routes.home}
          render={() => <Redirect to={routes.movies_app} />}
        />
        <Route path={routes.category} component={MoviesCards} />
        <Route path={routes.movieDetails} component={MovieDetails} />
        <Route path={routes.contest} component={Contest} />
        <Route path={routes.about} component={About} />
        <Route path={routes.contact} component={Contact} />
      </Switch>

      <Footer />
      <ScrollUpButton
        EasingType="easeInQuint"
        ShowAtPosition={300}
        AnimationDuration={1000}
      />
    </>
  </HashRouter>
);

export default App;
