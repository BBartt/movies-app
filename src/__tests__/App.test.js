import React from 'react';
import { shallow } from '../enzyme';
import App from '../App';

describe('Test <App /> component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders without crashing', () => {
    expect(wrapper).toBeTruthy();
  });
  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renders 1 <App /> component', () => {
    expect(wrapper).toHaveLength(1);
  });
});
