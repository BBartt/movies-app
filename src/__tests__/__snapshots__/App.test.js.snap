// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <App /> component match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <React.Fragment>
        <GlobalStyleComponent />
        <Header />
        <UNDEFINED />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Container)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            }
          }
          path="/movies-app"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Main)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            }
          }
          path="/movies-app"
        />
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(MoviesCards)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/category"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(MoviesDetails)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/movieDetails/:id"
          />
          <Route
            component={[Function]}
            path="/contest"
          />
          <Route
            component={[Function]}
            path="/about"
          />
          <Route
            component={[Function]}
            path="/contact"
          />
        </Switch>
        <Footer />
        <VerticleButton
          AnimationDuration={1000}
          EasingType="easeInQuint"
          ShowAtPosition={300}
        />
      </React.Fragment>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GlobalStyleComponent />,
          <Header />,
          <UNDEFINED />,
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Container)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/movies-app"
          />,
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Main)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/movies-app"
          />,
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(MoviesCards)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/category"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(MoviesDetails)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/movieDetails/:id"
            />
            <Route
              component={[Function]}
              path="/contest"
            />
            <Route
              component={[Function]}
              path="/about"
            />
            <Route
              component={[Function]}
              path="/contact"
            />
          </Switch>,
          <Footer />,
          <VerticleButton
            AnimationDuration={1000}
            EasingType="easeInQuint"
            ShowAtPosition={300}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(NavBar)",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Container)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
            "path": "/movies-app",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Main)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
            "path": "/movies-app",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(MoviesCards)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                path="/category"
              />,
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(MoviesDetails)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                path="/movieDetails/:id"
              />,
              <Route
                component={[Function]}
                path="/contest"
              />,
              <Route
                component={[Function]}
                path="/about"
              />,
              <Route
                component={[Function]}
                path="/contact"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(MoviesCards)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "path": "/category",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(MoviesDetails)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "path": "/movieDetails/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/contest",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/about",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/contact",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "AnimationDuration": 1000,
            "EasingType": "easeInQuint",
            "ShowAtPosition": 300,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <React.Fragment>
          <GlobalStyleComponent />
          <Header />
          <UNDEFINED />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Container)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/movies-app"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Main)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/movies-app"
          />
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(MoviesCards)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/category"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(MoviesDetails)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/movieDetails/:id"
            />
            <Route
              component={[Function]}
              path="/contest"
            />
            <Route
              component={[Function]}
              path="/about"
            />
            <Route
              component={[Function]}
              path="/contact"
            />
          </Switch>
          <Footer />
          <VerticleButton
            AnimationDuration={1000}
            EasingType="easeInQuint"
            ShowAtPosition={300}
          />
        </React.Fragment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GlobalStyleComponent />,
            <Header />,
            <UNDEFINED />,
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Container)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/movies-app"
            />,
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Main)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/movies-app"
            />,
            <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(MoviesCards)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                path="/category"
              />
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(MoviesDetails)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                path="/movieDetails/:id"
              />
              <Route
                component={[Function]}
                path="/contest"
              />
              <Route
                component={[Function]}
                path="/about"
              />
              <Route
                component={[Function]}
                path="/contact"
              />
            </Switch>,
            <Footer />,
            <VerticleButton
              AnimationDuration={1000}
              EasingType="easeInQuint"
              ShowAtPosition={300}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(NavBar)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Container)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "path": "/movies-app",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Main)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "path": "/movies-app",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
                />,
                <Route
                  component={
                    Object {
                      "$$typeof": Symbol(react.memo),
                      "WrappedComponent": [Function],
                      "compare": null,
                      "displayName": "Connect(MoviesCards)",
                      "type": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  path="/category"
                />,
                <Route
                  component={
                    Object {
                      "$$typeof": Symbol(react.memo),
                      "WrappedComponent": [Function],
                      "compare": null,
                      "displayName": "Connect(MoviesDetails)",
                      "type": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  path="/movieDetails/:id"
                />,
                <Route
                  component={[Function]}
                  path="/contest"
                />,
                <Route
                  component={[Function]}
                  path="/about"
                />,
                <Route
                  component={[Function]}
                  path="/contact"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(MoviesCards)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  },
                  "path": "/category",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(MoviesDetails)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  },
                  "path": "/movieDetails/:id",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/contest",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/about",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/contact",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "AnimationDuration": 1000,
              "EasingType": "easeInQuint",
              "ShowAtPosition": 300,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
